### PR TITLE
Replace Gopkg with go mod

### PR DESCRIPTION
**Pull request by ZhiminXiang**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,58 @@
+module knative.dev/net-certmanager
+
+go 1.14
```
Comment (chizhg) [inline go.mod:3]: Please keep it as `go 1.13`, we haven't started switching CI to use `go 1.14` yet.

File: hack/tools.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tools
+
+// This package imports things required by this repository, to force `go mod` to see them as dependencies
+import (
```
Comment (chizhg) [inline hack/tools.go:21]: I think only adding `_ "knative.dev/pkg/hack"` would suffice. Could you try removing all other and see if it'll work?

**Comment by chizhg** [comment]
To fix the integration tests:
You don't have to run the tests from `vendor` after this change, the tests will be just from another module. See https://github.com/knative/net-http01/blob/master/test/e2e-tests.sh#L23.

File: go.mod (hunk 1 above)
Comment (ZhiminXiang) [reply go.mod]: Done

File: hack/tools.go (hunk 2 above)
Comment (ZhiminXiang) [reply hack/tools.go:21]: looks like we don't need code-generator related imports. I think we still need knative.dev/test-infra/scripts https://github.com/knative/net-certmanager/blob/master/hack/release.sh#L20

**Comment by ZhiminXiang** [comment]
> To fix the integration tests:
> You don't have to run the tests from `vendor` after this change, the tests will be just from another module. See https://github.com/knative/net-http01/blob/master/test/e2e-tests.sh#L23.

Done

**Comment by chizhg** [comment]
It seems the `sed` command in https://github.com/knative/net-certmanager/blob/master/test/upload-test-images.sh#L33 can simply be dropped?

**Comment by ZhiminXiang** [comment]
hmm. looks like the _test.go file is not included when using "go mod"

File: hack/update-deps.sh [hunk 3]
```diff
@@ -23,33 +23,47 @@ set -o pipefail
...
 rm -rf $(find vendor/ -path '*/e2e/*_test.go')
 
+# Add permission for shell scripts
+chmod +x ./vendor/k8s.io/code-generator/generate-groups.sh
+chmod +x ./vendor/knative.dev/pkg/hack/generate-knative.sh
+
 # Do this for every package under "cmd" except kodata and cmd itself.
 # TODO(zhiminx): Uncomment once we have binaries.
-#update_licenses third_party/VENDOR-LICENSE "$(find ./cmd -type d | grep -v kodata | grep -vE 'cmd$')"
+#update_licenses third_party/VENDOR-LICENSE "$(find ./cmd -type d | grep -v kodata | grep -vE 'cmd$')"
```
Comment (mattmoor) [inline hack/update-deps.sh:69]: Add trailing newline:
```suggestion
#update_licenses third_party/VENDOR-LICENSE "$(find ./cmd -type d | grep -v kodata | grep -vE 'cmd$')"

```

**Comment by chizhg** [comment]
To fix the build tests, please add `export GO111MODULE=on` in `test/presubmit-tests.sh`

**Comment by ZhiminXiang** [comment]
> To fix the build tests, please add `export GO111MODULE=on` in `test/presubmit-tests.sh`

yea. just realized this. Done.

**Comment by ZhiminXiang** [comment]
OK. Looks like I need to use a newer version to make build test pass :)

**Comment by chizhg** [comment]
> OK. Looks like I need to use a newer version to make build test pass :)

Ah I missed that.. Do you want to change serving back to master? Otherwise let's get this in?

**Comment by ZhiminXiang** [comment]
> > OK. Looks like I need to use a newer version to make build test pass :)
> 
> Ah I missed that.. Do you want to change serving back to master? Otherwise let's get this in?

I think it is OK to get this in.

**Comment by chizhg** [comment]
> > > OK. Looks like I need to use a newer version to make build test pass :)
> > 
> > 
> > Ah I missed that.. Do you want to change serving back to master? Otherwise let's get this in?
> 
> I think it is OK to get this in.

Thanks a lot!!!
/lgtm

**Pull request merged**